File: editors/vscode/src/main/settings/ClassCheckMode.ts
export enum ClassCheckMode {
  None = "None",
  CheckClass = "Check class",
  CheckClassConstructor = "Check class constructor",
}

const FLAGS: Record<ClassCheckMode, string | null> = {
  [ClassCheckMode.None]: null,
  [ClassCheckMode.CheckClass]: "--check-class",
  [ClassCheckMode.CheckClassConstructor]: "--check-class-constructor",
};

export function classCheckFlag(mode: ClassCheckMode): string | null {
  return FLAGS[mode];
}

File: editors/intellij/src/main/kotlin/com/github/jshwi/docsig/ui/ClassCheckMode.kt
package com.github.jshwi.docsig.ui

import com.github.jshwi.docsig.messages.DocsigBundle

internal enum class ClassCheckMode(
    internal val label: String,
    internal val flag: String?,
) {
    NONE(
        DocsigBundle.message(
            "settings.options.class-checking-mode-none",
        ),
        null,
    ),
    CLASS(
        DocsigBundle.message(
            "settings.options.class-checking-mode-check-class",
        ),
        "--check-class",
    ),
    CLASS_CONSTRUCTOR(
        DocsigBundle.message(
            "settings.options.class-checking-mode-check-class-constructor",
        ),
        "--check-class-constructor",
    ),
}
